internal/services/logz/logz_sub_account_tag_rule_resource.go

```diff
@@ -105,32 +107,73 @@ func resourceLogzSubAccountTagRuleCreateUpdate(d *pluginsdk.ResourceData, meta i
 	}
 
 	id := tagrules.NewAccountTagRuleID(subAccountId.SubscriptionId, subAccountId.ResourceGroupName, subAccountId.MonitorName, subAccountId.AccountName, "default")
-	if d.IsNewResource() {
-		existing, err := client.SubAccountTagRulesGet(ctx, id)
-		if err != nil {
-			if !response.WasNotFound(existing.HttpResponse) {
-				return fmt.Errorf("checking for existing %s: %+v", id, err)
-			}
-		}
+
+	existing, err := client.SubAccountTagRulesGet(ctx, id)
+	if err != nil {
 		if !response.WasNotFound(existing.HttpResponse) {
-			return tf.ImportAsExistsError("azurerm_logz_sub_account_tag_rule", id.ID())
+			return fmt.Errorf("checking for existing %s: %+v", id, err)
 		}
 	}
+	if !response.WasNotFound(existing.HttpResponse) {
+		return tf.ImportAsExistsError("azurerm_logz_sub_account_tag_rule", id.ID())
+	}
 
 	payload := tagrules.MonitoringTagRules{
 		Properties: &tagrules.MonitoringTagRulesProperties{
-			LogRules: expandTagRuleLogRules(d),
+			LogRules: &tagrules.LogRules{
+				FilteringTags:        expandTagRuleFilteringTagArray(d.Get("tag_filter").([]interface{})),
+				SendAadLogs:          pointer.To(d.Get("send_aad_logs").(bool)),
+				SendSubscriptionLogs: pointer.To(d.Get("send_subscription_logs").(bool)),
+				SendActivityLogs:     pointer.To(d.Get("send_activity_logs").(bool)),
+			},
 		},
 	}
 
 	if _, err := client.SubAccountTagRulesCreateOrUpdate(ctx, id, payload); err != nil {
-		return fmt.Errorf("creating/updating %s: %+v", id, err)
+		return fmt.Errorf("creating %s: %+v", id, err)
 	}
 
 	d.SetId(id.ID())
 	return resourceLogzSubAccountTagRuleRead(d, meta)
 }
 
+func resourceLogzSubAccountTagRuleUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Logz.TagRuleClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
```

Comment:
_minor_ but this can become:

```suggestion
	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
```

(similar in the Create)